rest/handler.go

```diff
@@ -387,6 +400,40 @@ func (h *handler) checkAuth(context *db.DatabaseContext) (err error) {
 	return nil
 }
 
+func (h *handler) checkAdminAuth(dbContext *db.DatabaseContext, checkPermissions []string, storePermissionsResults bool) error {
+	var managementEndpoints []string
+	var err error
+	var httpClient *http.Client
+
+	if dbContext != nil {
+		managementEndpoints, httpClient, err = dbContext.ObtainManagementEndpointsAndHTTPClient()
+		if err != nil {
+			return base.HTTPErrorf(http.StatusInternalServerError, "")
+		}
+	} else {
+		managementEndpoints, httpClient, err = h.server.ObtainManagementEndpointsAndHTTPClient()
+		if err != nil {
+			return base.HTTPErrorf(http.StatusInternalServerError, "")
+		}
+	}
+
+	username, password := h.getBasicAuth()
+	statusCode, permResults, err := h.server.CheckPermissions(httpClient, managementEndpoints, username, password, checkPermissions)
+	if err != nil {
+		return base.HTTPErrorf(http.StatusInternalServerError, "")
+	}
```

Comment:
At what point do we actually get information about any errors?

None of this code is doing any logging at all - which is OK if the error is being bubbled up enough to where it is being logged, but it doesn't look like that's happening.

Reply:
Agreed that we don't want to expose to end user, but definitely want logging on security failure (and possibly success).

Reply:
Opted to bubble up errors to where we call `checkAdminAuth` and at that point we do logging. In the case of an error log the error, in the case of a not-OK status code, log the failure and return status code to user otherwise log successful login.